src/python/pants/backend/google_cloud_function/python/rules_test.py

```diff
@@ -191,48 +194,78 @@ def handler(event, context):
 
 
 @pytest.mark.parametrize(
-    ("ics", "runtime"),
+    ("ics", "runtime", "complete_platforms_target_type"),
     [
-        pytest.param(["==3.7.*"], None, id="runtime inferred from ICs"),
-        pytest.param(None, "python37", id="runtime explicitly set"),
+        pytest.param(["==3.7.*"], None, None, id="runtime inferred from ICs"),
+        pytest.param(None, "python37", None, id="runtime explicitly set"),
+        pytest.param(["==3.7.*"], None, "file", id="complete platforms with file target"),
+        pytest.param(["==3.7.*"], None, "resource", id="complete platforms with resource target"),
```

Comment:
Thanks for writing some tests!

Given the observation above about this applying to more than just GCF... I wonder whether it might be good to test the `digest_complete_platform_addresses` rule more directly (and maybe `pex_binary` directly), and assume that all the derived targets work similarly.

What do you think?

Reply:
Certainly possible! I defaulted to setting up these tests for GCF because there was already some scaffolding in place for it which made it easy to add on to. 

I'll admit I'm still a bit flummoxed as to why the `pex_binary` target worked even without this change, perhaps adding some tests could help uncover the reason why.

Reply:
Added some more tests sprinkled around the codebase that should cover all of these cases...have a look and let me know what you think.